modules/reindex/src/test/java/org/elasticsearch/reindex/AsyncBulkByScrollActionTests.java

```diff
@@ -826,6 +876,30 @@ public void done(TimeValue extraKeepAlive) {}
         expectThrows(IllegalArgumentException.class, () -> response.consumeHits(1));
     }
 
+    public void testEnableScrollByDefault() {
+        var preparedSearchRequest = AbstractAsyncBulkByScrollAction.prepareSearchRequest(testRequest, false, false);
+        assertThat(preparedSearchRequest.scroll(), notNullValue());
+    }
+
+    public void testDisableScrollWhenMaxDocsIsLessThenScrollSize() {
+        testRequest.setMaxDocs(1);
+        testRequest.getSearchRequest().source().size(100);
+
+        var preparedSearchRequest = AbstractAsyncBulkByScrollAction.prepareSearchRequest(testRequest, false, false);
+
+        assertThat(preparedSearchRequest.scroll(), nullValue());
+    }
```

Comment:
Can we add a test verifying that a max-docs set above the scroll size has scroll enabled (like max_docs=101 for scroll_size=100).